### ISSUE TITLE
windows_ad_join resource appears broken in Chef 14.5.27

### ISSUE THREAD

**Issue by cpjones01** [chef/chef#7672]
## Description
Using the windows_ad_join resource generates produces an "undefined method `new_name' for Chef::Resource::WindowsAdJoin" error.

## Chef Version
14.5.27

## Platform Version
Windows Serer 2012r2

## Replication Case
```
windows_ad_join node['ad']['domain_name'] do
  domain_user join_creds['username']
  domain_password join_creds['pass']
  ou_path node['ad']['ou_path']
end
```

## Client Output

```
* windows_ad_join[datacenter] action join

           ================================================================================
           Error executing action `join` on resource 'windows_ad_join[datacenter]'
           ================================================================================

           NoMethodError
           -------------
           undefined method `new_name' for Chef::Resource::WindowsAdJoin

           Resource Declaration:
           ---------------------
           suppressed sensitive resource output

           Compiled Resource:
           ------------------
           suppressed sensitive resource output

           System Info:
           ------------
           chef_version=14.5.27
           platform=windows
           platform_version=6.3.9600
           ruby=ruby 2.5.1p57 (2018-03-29 revision 63029) [x64-mingw32]
           program_name=C:/opscode/chef/bin/chef-client
           executable=C:/opscode/chef/bin/chef-client
```

**Comment by tas50** [comment]
Fixed by https://github.com/chef/chef/pull/7673. We'll get a regression release out today. Thanks for the report.

**Comment by tas50** [comment]
This was released in a hotfix for Chef 14.5 Thanks for the report.
